litex/build/efinix/common.py

```diff
@@ -261,13 +244,130 @@ class EfinixDifferentialInput:
     def lower(dr):
         return EfinixDifferentialInputImpl(dr.platform, dr.i_p, dr.i_n, dr.o)
 
+
+
+
+
+# Efinix DDRTristate ---------------------------------------------------------------------------------
+
+class EfinixDDRTristateImpl(Module):
+    def __init__(self, platform, io, o1, o2, oe1, oe2, i1, i2, clk):
+        assert oe1 == oe2
```

Comment:
this assert doesn't work right if `oe1` and `oe2` are a `_Slice`, like in the litespi ddr phy

Reply:
If i remember well, one issue is that the efinix primitive only support 1 output enable signal (in ddr mode).
- Either when oe1 != oe2 it could fall back to DDR tristate emulation (et get very bad timings)
- Either it could be ignored and hope for the best
- Either it can keep the assert

Not sure what is the best.

Reply:
Another options would be to update litespi to avoid the usage of 2 oe signals

Reply:
it was already with changes like: https://github.com/litex-hub/litespi/pull/60

The problem is this: `oe1 = dq_oe[i], oe2 = dq_oe[i],` with `dq_oe = Signal(len(pads.dq))`
where `oe1` and `oe2` are only a `_Slice``od a Signal and not a whole Signal. the error comes from here: https://github.com/m-labs/migen/blob/832a7240ba32af9cbd4fdd519ddcb4f912534726/migen/fhdl/structure.py#L41

Reply:
this should fix it https://github.com/m-labs/migen/pull/292